plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -104,7 +105,7 @@
     private final MachineRuntimeInfoImpl                      machineRuntime;
 
     @Inject
-    public DockerInstance(DockerConnector docker,
+    public DockerInstance(DockerConnectorProvider dockerProvider,
```

Comment:
@benoitf BTW, similar approach is used for obtaining ServerEvaluationStrategy (default / docker-local) -  https://github.com/eclipse/che/blob/master/plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java#L52
And I think it is relevant to also use MapBinder for DockerConnector use-case (default / openshift)